### PR TITLE
TYP: fix type ignores

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

File: pandas/compat/numpy/__init__.py [hunk 1]
```diff
@@ -27,41 +27,39 @@
...
-    s = tz_replacer(s)
-    # error: No overload variant of "datetime64" matches argument types "Any",
-    # "Tuple[Any, ...]", "Dict[str, Any]"
-    return np.datetime64(s, *args, **kwargs)  # type: ignore[call-overload]
+    tstring = _tz_replacer(tstring)
+    return np.datetime64(tstring, unit)
 
 
-def np_array_datetime64_compat(arr, *args, **kwargs):
+def np_array_datetime64_compat(arr, dtype="M8[ns]"):
```
Comment (jreback) [inline pandas/compat/numpy/__init__.py:49]: btw I think we can blow all of this compat code away :->

**Review by jreback — APPROVED** [review]
lgtm. @simonjayhawkins

File: pandas/compat/numpy/__init__.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/compat/numpy/__init__.py:49]: will take a look in the next pass

**Review by simonjayhawkins — APPROVED** [review]
Thanks @jbrockmendel lgtm

**Pull request merged**